h2/src/main/org/h2/mvstore/tx/TransactionMap.java

```diff
@@ -553,13 +550,20 @@ private void fetchNext() {
 
             @Override
             public boolean hasNext() {
+                if(current == null) {
+                    fetchNext();
+                }
```

Comment:
this is on a hot path, why add extra instructions here instead of initialinsg current in the constructor?